src/indigo/components/SearchBar.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {ButtonToolbar, FormControl} from 'react-bootstrap';
+import Icon from './icons'
+import PropTypes from 'prop-types';
+
+
+class SearchBar extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {query: this.props.query};
+    this.changeQuery = this.changeQuery.bind(this)
+    this.submitQuery = this.submitQuery.bind(this)
+  }
+  
+  changeQuery(event) {
+    this.setState({query: event.target.value})
+  }
+  
+  submitQuery(event) {
+    event.preventDefault();
+    this.props.onSubmit(this.state.query);
+  }
+  
+  render() {
+    return (
+      <form className="searchbar" onSubmit={(event) => this.submitQuery(event)}>
+        <Icon.Search className="searchbar-icon icon-size-20"/>
+        <FormControl
+          value={this.state.query}
+          placeholder={this.props.placeholder}
+          onChange={(event) => this.changeQuery(event)}
+          className="searchbar-input"
+          ref="searchInput"
+          type="text"
+          autoFocus
+        />
+        {!!this.props.actions &&
+        <ButtonToolbar className="searchbar-actions">
+          {this.props.actions}
+        </ButtonToolbar>
+        }
+      </form>
+    );
+  }
+}
+
+SearchBar.defaultProps = {
+  onChange: () => {
+  },
+  onSubmit: (query) => query,
+  placeholder: 'Search'
+};
+
+
+SearchBar.propTypes = {
+  query: PropTypes.string.isRequired,
+  onChange: PropTypes.func,
+  onSubmit: PropTypes.func,
+  className: PropTypes.string,
+  placeholder: PropTypes.string,
+  actions: PropTypes.object
```

Comment:
co presne budu robit `actions`? z prikladu vidim, ze to bude prijimat react komponentu ... dospecifikuj teda lepsie,  co to bude ... `object` je dost vseobecne. Resp. kludne tu property premenuj.

Este mi napada, ci by neslo nejako vyzadovat, ze ten button co tam posles by bol `type=submit`. 

A co keby sme v tom neboli taki benevolentni na nedovolili tam dat vlastny submit button? Alebo mu povolili zmenit len label? Neviem,no :)

Reply:
Nemusi ty byt nutne tlacitko 'search'. Ma to vicemene layoutovat akce vedle searchbaru. Napr v Job List, nebo Trash. Prejmenoval jsem vystizneji. Jeste jsem to trochu zmenil a rozsiril